state/accumulator.go

```diff
@@ -238,7 +238,7 @@ func (a *Accumulator) Initialise(roomID string, state []json.RawMessage) (Initia
 				Str("room_id", roomID).
 				Int("len_state", len(events)).
 				Msg(errMsg)
-			return fmt.Errorf(errMsg)
+			return internal.NewDataError(errMsg)
```

Comment:
To check: this means "the homeserver has given us duff data and there's no point retrying this"?

Reply:
Might be worth a comment to this effect as on +392 below